### ISSUE TITLE
Extending template does not seem to work

### ISSUE THREAD

**Issue by jlumbroso** [josephlou5/roseingrave#21]
I tried to add the key `endSignsX` to the template, by adding it to the `metaDataFields` or `template-definitions.json`:

```
  "metaDataFields": {
    "title": "Title",
    "tempo": "Tempo",
    "isPianoClefs": "Clefs G2 and F4?",
    "clefs": "Clefs other than G2 and F4",
    "keySig": "Key signature",
    "timeSig": "Time signature",
    "notBipartite": "Sonata not bipartite",
    "barCount": "Number of bars (1st half + 2nd half)",
    "compassHigh": "Highest note",
    "compassLow": "Lowest note",
    "hand": "Hand signs",
    "endSignsA": "Endings signs (1st half)",
    "endSignsB": "Endings signs (2nd half)",
    "endSignsX": "Additional endings",    /* <------- here */
    "repeatSigns": "Repeat signs other than endings",
    "articulation": "Articulation signs (slurs, staccato, etc.)",
    "dynamic": "Dynamic signs",
    "ornamentTrill": "Ornamentation: trills /\\/\\?",
    "ornamentPlus": "Ornamentation: + signs?",
    "ornamentTr": "Ornamentation: Tr.?",
    "ornamentAcciaccatura": "Acciaccaturas? (barred/slashed grace note)",
    "ornamentAppoggiatura": "Appoggiaturas? (grace note)",
    "ornamentTremulo": "Trem. or Tremulo?",
    "ornamentOthers": "Other ornamentations",
    "otherIndications": "Other indications"
  },
```

When running Roseingrave, two things happen:

- It complains that the key does not exist (which it means to say that this is not in the default template, as reported by: https://github.com/scarlatti/roseingrave/blob/c77c8a37a0cd16a101588ef6c116d81231f3c2ae/src/roseingrave/_input_files.py#L97-L102
![Screenshot 2023-02-23 at 8 43 52 PM](https://user-images.githubusercontent.com/11770469/221072035-458e02ed-bf7d-4666-aa1d-12e6f337fcad.png)
- But the above code suggests it should work anyway, no? Yet the sheet generated does not contain the entry:
![Screenshot 2023-02-23 at 8 47 38 PM](https://user-images.githubusercontent.com/11770469/221072007-6a53efbb-479f-4cfc-8006-518c1794154e.png)

Questions:
- In the current state of affairs is currently possible (supported) to extend the template through `template-definition.json` or can it only support existing fields?
- Is this a bug?

**Comment by josephlou5** [comment]
Interestingly, it is _not_ supported to add additional `"metaDataFields"` through your own definition file; you must also define the key-value pair in the default JSON file. As you noticed, unknown keys will be skipped, and the resulting sheet will not contain it. This indeed seems like a bug, since there is no reason to constrain the user to only the fields defined in the defaults file. In fact, this defaults file hardcodes a lot of information that probably shouldn't be hardcoded. Because of this hardcoding, I had never thought to test any other settings, and thus ended up with this bug.

This seems like a reasonable feature to add.

**Comment by josephlou5** [comment]
This will be included in the v1.0.0 update.
